pkg/state/state_sriov_dp_test.go

```diff
@@ -14,130 +14,280 @@ See the License for the specific language governing permissions and
+		renderer = r
+		catalog = getTestCatalog()
+		namespace = config.FromEnv().State.NetworkOperatorResourceNamespace
+	})
 
-			files, err := utils.GetFilesWithSuffix(manifestBaseDir, render.ManifestFileSuffix...)
+	Context("When creating NCP with SRIOV-device-plugin", func() {
+		It("should create Daemonset - minimal spec", func() {
+			By("Sync")
+			cr := getMinimalNicClusterPolicyWithSriovDp(false)
+			status, err := sriovDpState.Sync(context.Background(), cr, catalog)
 			Expect(err).NotTo(HaveOccurred())
-			renderer := render.NewRenderer(files)
-
-			stateName := "state-SRIOV-device-plugin"
-			sriovDpState := stateSriovDp{
-				stateSkel: stateSkel{
-					name:        stateName,
-					description: "SR-IOV device plugin deployed in the cluster",
-					client:      &client,
-					renderer:    renderer,
-				},
+			Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+			By("Verify DaemonSet")
+			ds := &appsv1.DaemonSet{}
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
+			// expect privileged mode
+			Expect(*ds.Spec.Template.Spec.Containers[0].SecurityContext.Privileged).To(BeTrue())
+			assertSriovDpPodTemplatesVolumeFields(&ds.Spec.Template, false)
+			assertSriovDpPodTemplatesVolumeMountFields(&ds.Spec.Template, false)
+		})
+		It("should create Daemonset with CDI support when specified in CR", func() {
+			By("Sync")
+			cr := getMinimalNicClusterPolicyWithSriovDp(true)
+			status, err := sriovDpState.Sync(context.Background(), cr, catalog)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+			By("Verify DaemonSet")
+			ds := &appsv1.DaemonSet{}
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
+			// expect privileged mode
+			Expect(*ds.Spec.Template.Spec.Containers[0].SecurityContext.Privileged).To(BeTrue())
+			assertSriovDpPodTemplatesVolumeFields(&ds.Spec.Template, true)
+			assertSriovDpPodTemplatesVolumeMountFields(&ds.Spec.Template, true)
+		})
+	})
+	Context("Verify Sync flows", func() {
+		It("should create Daemonset, update state to Ready", func() {
+			By("Sync")
+			cr := getMinimalNicClusterPolicyWithSriovDp(false)
+			status, err := sriovDpState.Sync(context.Background(), cr, catalog)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+			By("Verify DaemonSet")
+			ds := &appsv1.DaemonSet{}
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
```

Comment:
nit: we dont really need to re check what was checked in previous cases. just the status here imo.
that said. up to u.